File: programmers/lv0/코딩기초트레이닝/조건문/이어붙인수/Solution.kt
// 1
class Solution {
    data class NumObj(
        val evens: MutableList<Int> = mutableListOf(),
        val odds: MutableList<Int> = mutableListOf(),
    )

    fun solution(num_list: IntArray): Int {
        val (evens, odds) = num_list.fold(NumObj()) { acc, num ->
            if (num % 2 == 0) acc.evens.add(num)
            else acc.odds.add(num)
            acc
        }
        return evens.joinToString(separator = "").toInt() + odds.joinToString(separator = "").toInt()
    }
}

// 2
class Solution {
    fun solution(numList: IntArray) = numList.partition { it % 2 == 0 }.toList().sumOf { it.joinToString("").toInt() }
}

File: programmers/lv0/코딩기초트레이닝/조건문/이어붙인수/Solution.py
def solution(num_list):
    evens = ''.join(str(num) for num in num_list if num % 2 == 0)
    odds = ''.join(str(num) for num in num_list if num % 2 != 0)
    return int(evens or '0') + int(odds or '0')